Source/Core/VideoCommon/TextureDecoder_x64.cpp

```diff
@@ -1486,6 +1487,52 @@ void _TexDecoder_DecodeImpl(u32* dst, const u8* src, int width, int height, Text
     TexDecoder_DecodeImpl_CMPR(dst, src, width, height, texformat, tlut, tlutfmt, Wsteps4, Wsteps8);
     break;
 
+  case TextureFormat::XFB:
+  {
+    for (int y = 0; y < height; y += 1)
+    {
+      for (int x = 0; x < width; x += 2)
+      {
+        size_t offset = (y * width + x) * 2;
```

Comment:
This is performing sign-conversions without an explicit cast. Just to point that out.